subsys/bluetooth/host/id.c

```diff
@@ -1281,20 +1281,31 @@ int bt_id_create(bt_addr_le_t *addr, uint8_t *irk)
 {
 	int new_id, err;
 
-	if (addr && !bt_addr_le_eq(addr, BT_ADDR_LE_ANY)) {
-		if (addr->type != BT_ADDR_LE_RANDOM ||
-		    !BT_ADDR_IS_STATIC(&addr->a)) {
-			LOG_ERR("Only static random identity address supported");
-			return -EINVAL;
-		}
+	if (!IS_ENABLED(CONFIG_BT_PRIVACY) && irk) {
+		return -EINVAL;
+	}
 
+	if (addr && !bt_addr_le_eq(addr, BT_ADDR_LE_ANY)) {
 		if (id_find(addr) >= 0) {
 			return -EALREADY;
 		}
-	}
 
-	if (!IS_ENABLED(CONFIG_BT_PRIVACY) && irk) {
-		return -EINVAL;
+		if (addr->type == BT_ADDR_LE_PUBLIC) {
+			/* set the single public address */
+			bt_addr_le_t *default_id = &bt_dev.id_addr[BT_ID_DEFAULT];
+
+			if (bt_dev.id_count > 0 && !bt_addr_le_eq(default_id, BT_ADDR_LE_ANY)) {
```

Comment:
I don't think you need to check for `BT_ADDR_LE_ANY` here, since we don't allow creating identities with that address. Or to put it another way, if `id_count > 0` then the default id has to be some valid address rather than `BT_ADDR_LE_ANY`. If you look at `id.c` and the `id_create()` function, you'll see that if it gets `BT_ADDR_LE_ANY` it branches off to generate a static random address instead.

Reply:
It could theoretically have been deleted by `bt_id_delete()`, which sets it to `BT_ADDR_LE_ANY`, but only decrements `id_count` if it was the last id. Probably a very niche edge case, I can remove the check if it adds more confusion than it solves.

Reply:
Or is there other some use case I'm not seeing here?

Reply:
> It could theoretically have been deleted by `bt_id_delete()`, which sets it to `BT_ADDR_LE_ANY`, but only decrements `id_count` if it was the last id. Probably a very niche edge case, I can remove the check if it adds more confusion than it solves.

I think we should probably prohibit deleting a public address identity.

Reply:
Oh, actually that's already the case, `bt_id_delete()` forbids deleting the default ID. I've applied your suggestion of `(bt_dev.id_count != 0)`.